src/lib/cloud-provider.js

```diff
@@ -65,6 +58,26 @@ class CloudProvider {
         };
     }
 
+    parseMessage (message) {
+        const varData = {};
+        switch (message.method) {
+        case 'ack': {
+            varData.varCreate = {
+                name: message.name
+            };
+            break;
+        }
+        case 'set': {
+            varData.varUpdate = {
+                name: message.name,
+                value: message.value
+            };
+            break;
+        }
+        }
+        this.vm.postIOData('cloud', varData);
```

Comment:
Can you return the varData and put the postIOData in the caller of this function? Seems strange to have a method called "parse_" that actually sends data to the VM.